impl/src/main/java/org/ehcache/internal/store/offheap/OffHeapStore.java

```diff
@@ -141,7 +138,7 @@ public OffHeapStore(final Configuration<K, V> config, Serializer<K> keySerialize
         public OffHeapValueHolder<V> apply(K mappedKey, OffHeapValueHolder<V> mappedValue) {
           long now = timeSource.getTimeMillis();
 
-          if (mappedValue == null || mappedValue.isExpired(now)) {
+          if (mappedValue == null || mappedValue.isExpired(now, OffHeapValueHolder.TIME_UNIT)) {
```

Comment:
Indeed, in this particular case, the should be TimeUnit.MILLISECONDS and not OffHeapValueHolder.TIME_UNIT since the 'now' timestamp is coming from the timesource and not from the offheap store's value holder.

If you're speaking only about the isExpired() method, you're right that the time unit should always be TimeUnit.MILLISECONDS which makes me believe that we could get rid of this extra parameter.